src/main/scala/org/neo4j/spark/dataframe/Neo4jDataFrame.scala

```diff
@@ -206,13 +157,13 @@ object Neo4jDataFrame {
 
     override def compute(split: Partition, context: TaskContext): Iterator[Row] = {
       val driver = config.driver()
-      val session = driver.session()
+      val session = driver.session(config.sessionConfig())
       try {
-        val runTransaction = new TransactionWork[Iterator[Row]]() {
+        val txWork = new TransactionWork[Iterator[Row]] {
           override def execute(tx: Transaction): Iterator[Row] = {
             val result = tx.run(query, params)
             if (!result.hasNext) throw new RuntimeException("Can't determine schema from empty result")
-            result.asScala.map(record => {
+            result.list().asScala.map(record => {
```

Comment:
this is potentially a lot that's eagerly pulled ... (think 10 or 100M or 1bn rows ...) ?
I meant to defer the materialization as the iterator is used by spark
and close resources after the last record has been consumed.

Reply:
ping?